src/System.Device.Gpio/System/Device/ComponentInformation.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+#pragma warning disable CS1591
+namespace System.Device
+{
+    public record ComponentInformation
+    {
+        private readonly List<ComponentInformation> _subComponents;
+
+        public ComponentInformation(object instance, string name)
+            : this(instance, name, ComponentState.Active)
+        {
+        }
+
+        public ComponentInformation(object instance, string name, ComponentState componentState)
```

Comment:
can we skip ComponentState for this initial review?

Reply:
done